src/main/scala/example/ConfigMixins.scala

```diff
@@ -0,0 +1,170 @@
+package example
+
+import chisel3._
+import freechips.rocketchip.config.{Parameters, Config}
+import freechips.rocketchip.subsystem.{WithRoccExample, WithNMemoryChannels, WithNBigCores, WithRV32}
+import freechips.rocketchip.diplomacy.{LazyModule, ValName}
+import freechips.rocketchip.devices.tilelink.BootROMParams
+import freechips.rocketchip.tile.XLen
+import testchipip._
+import sifive.blocks.devices.gpio._
+
+/**
+ * TODO: Why do we need this?
+ */
+object ConfigValName {
+  implicit val valName = ValName("TestHarness")
+}
+import ConfigValName._
+
+// -----------------------
+// Common Parameter Mixins
+// -----------------------
+
+/**
+ * Class to specify where the BootRom file is (from `rebar` top)
+ */
+class WithBootROM extends Config((site, here, up) => {
+  case BootROMParams => BootROMParams(
+    contentFileName = s"./bootrom/bootrom.rv${site(XLen)}.img")
+})
+
+/**
+ * Class to add in GPIO
+ */
+class WithGPIO extends Config((site, here, up) => {
+  case PeripheryGPIOKey => List(
+    GPIOParams(address = 0x10012000, width = 4, includeIOF = false))
+})
+
+// ----------------------------------------
+// Rocket Top Level System Parameter Mixins
+// ----------------------------------------
+
+/**
+ * Class to specify a "plain" top level rocket-chip system
+ */
+class WithNormalRocketTop extends Config((site, here, up) => {
```

Comment:
Why can't the generator inject these into the parameter space? Do you guys just specify a config, not a top-level module in your generator arguments?

Reply:
Thats an option that would remove all of these mixins, but we have to do the connections somehow (aka the `dut.connectIO`). Im assuming that would still be done in the generator?

Reply:
Also, I think that can be punted until another PR?

Reply:
I warming to the idea of having methods like that in our targets (see the comment in my PR). Eventually, i think we'd want to do the same thing with firesim endpoints.

Reply:
So, I think this can be punted down the road. And an issue can be made of it for followup.

Reply:
Offline: Merging in. Dealing with this later.